src/nidmm/system_tests/test_system_nidmm.py

```diff
@@ -244,3 +244,64 @@ def test_configure_waveform_acquisition(session):
     assert session.range == 100
     assert session.waveform_rate == 100000
     assert session.waveform_points == 400
+
+
+def test_fetch_waveform(session):
+    session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 10, 1000, 10)
+    with session.initiate():
+        number_of_points_to_read = 10
+        measurements,actual_number_of_points = session.fetch_waveform(number_of_points_to_read)
+        assert len(measurements) == number_of_points_to_read
+        assert actual_number_of_points == number_of_points_to_read
+
+
+def test_fetch_waveform_error(session):
+    try:
+        session.configure_waveform_acquisition(nidmm.Function.WAVEFORM_VOLTAGE, 10, 1000, 10)
+        with session.initiate():
+            number_of_points_to_read = 100
+            session.fetch_waveform(number_of_points_to_read)   # trying to fetch points more than configured
+    except nidmm.Error as e:
```

Comment:
You should 'assert False' after the fetch_waveform since you expect an error. That way you don't get a false pass if it doesn't actually error.